sklearn/utils/tests/test_stats.py

```diff
@@ -70,20 +74,146 @@ def test_weighted_percentile_2d():
     x2 = rng.randint(20, size=10)
     x_2d = np.vstack((x1, x2)).T
 
-    w_median = _weighted_percentile(x_2d, w1)
+    w_median = _weighted_percentile(x_2d, w1, interpolation=interpolation)
     p_axis_0 = [
-        _weighted_percentile(x_2d[:, i], w1)
+        _weighted_percentile(x_2d[:, i], w1, interpolation=interpolation)
         for i in range(x_2d.shape[1])
     ]
     assert_allclose(w_median, p_axis_0)
 
-    # Check when array and sample_weight boht 2D
+    # Check when array and sample_weight both 2D
     w2 = rng.choice(5, size=10)
     w_2d = np.vstack((w1, w2)).T
 
-    w_median = _weighted_percentile(x_2d, w_2d)
+    w_median = _weighted_percentile(x_2d, w_2d, interpolation=interpolation)
     p_axis_0 = [
-        _weighted_percentile(x_2d[:, i], w_2d[:, i])
+        _weighted_percentile(x_2d[:, i], w_2d[:, i],
+                             interpolation=interpolation)
         for i in range(x_2d.shape[1])
     ]
     assert_allclose(w_median, p_axis_0)
+
+
+def test_weighted_percentile_np_median():
+    # check that our weighted percentile lead to the same results than
+    # unweighted NumPy implementation with unit weights for the median
+    rng = np.random.RandomState(42)
+    X = rng.randn(10)
+    X.sort()
+    sample_weight = np.ones(X.shape)
+
+    np_median = np.median(X)
+    sklearn_median = _weighted_percentile(
+        X, sample_weight, percentile=50.0, interpolation="linear"
+    )
+
+    assert sklearn_median == pytest.approx(np_median)
```

Comment:
doesn't the next test include this one?

Reply:
This one check `np.median` while the next test check `np.percentile`

Reply:
There is no option for interpolation in `np.median`